core/src/main/java/tc/oc/pgm/rage/RageMatchModule.java

```diff
@@ -28,13 +29,24 @@ public void handlePlayerDamage(EntityDamageByEntityEvent event) {
   }
 
   private boolean isRage(Entity damager) {
-    if (damager instanceof Player) {
-      Player player = (Player) damager;
+    if (damager instanceof Player player) {
       return player.getItemInHand().containsEnchantment(Enchantments.SHARPNESS);
-    } else if (damager instanceof Arrow) {
-      Arrow arrow = (Arrow) damager; // Arrows with damage > 2 are from power bows.
-      return arrow.getShooter() instanceof Player && MISC_UTILS.getArrowDamage(arrow) > 2.0D;
+    } else if (damager instanceof Projectile proj) {
+      if (!(proj.getShooter() instanceof Player)) return false;
+      if (Platform.isModern()) {
+        // In modern Minecraft, we have access to the item stack that fired the projectile.
+        // We can leverage that to figure out if the projectile comes from a rage weapon or not
+        var item = MISC_UTILS.getProjectileWeapon(proj);
+        return item != null && item.containsEnchantment(Enchantments.POWER);
+      } else {
+        // On 1.8 the arrow damage takes enchantments into account
+        if (damager instanceof Arrow arrow) {
+          // Arrows with damage > 2 are from power bows.
+          return MISC_UTILS.getArrowDamage(arrow) > 2.0D;
+        }
```

Comment:
the whole point of platform utils etc is to hide away platform-dependant code, ideally you'd just have something like:
```suggestion
      return MISC_UTILS.hasPowerEnchant(proj);
```
and let the different implementations figure out how that is done, legacy can do arrow damage > 2.0D, modern can do the get-item and check for power